fix(Chat): Correctly initialize socket connection for chat system

nwjbrandon opened the pull request:
Previously the socket connection is initialize in the socket context.
Socket is not initialized at the conversation page and a null error
is shown.

This commit ensure that whenever the chat is loaded the socket
connection is completed first before emitting any services

nwjbrandon committed:
fix(Chat): Correctly initialize socket connection for chat system

Previously the socket connection is initialize in the socket context.
Socket is not initialized at the conversation page and a null error
is shown.

This commit ensure that whenever the chat is loaded the socket
connection is completed first before emitting any services

karrui commented on src/routes/chat/Chat.jsx:
```diff
@@ -1,8 +1,9 @@
 import PageContainer from 'components/pageContainer';
-
+import { useSocket } from 'contexts/socketContext';
+import { fetchChatRoomAction, fetchChatListAction } from 'reducers/ChatDux';
```
Nit: Don't need the word `action`

karrui commented on src/routes/chat/Chat.jsx:
```diff
@@ -34,10 +35,22 @@ const ChatContent = () => {
 const Chat = () => {
+  const dispatch = useDispatch();
   const { chatRoomId } = useParams();
   const correspondentName = useSelector(
```
Repeating myself from the other review: we don't need this variable now since we don't have the dealerName anymore.

nwjbrandon replied:
Wanted to fix this in next PR
But ok

nwjbrandon replied:
I wanted to resolve this in the next PR
But ok

karrui commented on src/routes/chat/Chat.jsx:
```diff
@@ -34,10 +35,22 @@ const ChatContent = () => {
   );
+  const { openChatSocket, closeChatSocket } = useSocket();
+  useEffect(() => {
+    openChatSocket();
```
Repeating: we don't initialize this here, we should initialize them in `AuthenticatedApp` component, and close the socket there too

karrui commented on src/services/SocketService/socketRequestService.js:
```diff
@@ -3,7 +3,7 @@ import tokenUtils from 'utils/tokenUtils';
 
 const requestChatList = () => {
-  Socket.socket.emit('set_chat_list', {
+  Socket.socket.emit('req_chat_rooms', {
```
Put `'req_chat_rooms'` in constants/sockets.js and then refer it here. Do the same for all channels.

karrui reviewed (changes requested):
Must easier to review. Just fix these changes.

karrui commented:
Please don't merge directly to master. Please merge to `staging`.

nwjbrandon committed:
refactor(chat): rename redux actions and move socket event to a file

This patch rename redux actions by removing the suffix action in
the names. The socket events are also move to socket.js under constants
directory.

nwjbrandon committed:
refactor(Chat): Change redux and file names

This patch gets rids of poor naming of redux store variables and
file names. ChatRoom is renamed to ChatConversation. ChatList is
renamed to ChatRooms. Prefix such as get, set, add, insert, are
used in the redux action.

nwjbrandon committed:
refactor(chat): move socket config to context and rename socket functions

This patch moves the socket connection to the socket context. The socket
function names are renamed.

nwjbrandon committed:
refactor(socket): wrap socket provider over authenticated app in app.js

karrui reviewed (approved):
LGTM! Congrats! HAHA
